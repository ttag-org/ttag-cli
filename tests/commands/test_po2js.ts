import * as path from "path";
import { execSync } from "child_process";

const poPath = path.resolve(__dirname, "../fixtures/po2jsTest/po2js.po");

test("convert po to js", () => {
    const result = execSync(
        `ts-node src/index.ts po2json ${poPath}`
    ).toString();
    expect(result).toMatchSnapshot();
});

test("convert po to js pretty", () => {
    const result = execSync(
        `ts-node src/index.ts po2json -p ${poPath}`
    ).toString();
    expect(result).toMatchSnapshot();
});

test("convert po to js nostrip", () => {
    const result = execSync(
        `ts-node src/index.ts po2json -n ${poPath}`
    ).toString();
    expect(result).toMatchSnapshot();
});

test("should apply compact format", () => {
    const result = execSync(
        `ts-node src/index.ts po2json --format=compact -n ${poPath}`
    ).toString();
    const jsonResult = JSON.parse(result);
    expect(jsonResult).toHaveProperty("contexts");
    expect(result).toMatchSnapshot();
});

const errMessage =
    'Duplicate msgid ("test ${ num2 }" and "test ${ num1 }"' +
    ' in the same context will be interpreted as the same key "test ${0}")' +
    " this potentially can lead to translation loss.";

const poPath2 = path.resolve(__dirname, "../fixtures/checkTest/same_key.po");
test("Should get exception about same key", () => {
    try {
        execSync(`ts-node src/index.ts po2json --format=compact -n ${poPath2}`);
        expect(false).toBe(true); // must fail anyway
    } catch (err) {
        expect(err.status).toBe(1);
        expect(err.stderr.toString()).toContain(errMessage);
    }
});
