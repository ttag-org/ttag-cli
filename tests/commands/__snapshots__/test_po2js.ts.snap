// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert po to js 1`] = `"{\\"charset\\":\\"utf-8\\",\\"headers\\":{\\"Content-Type\\":\\"text/plain; charset=utf-8\\",\\"Plural-Forms\\":\\"nplurals=2; plural=(n!=1);\\"},\\"translations\\":{\\"\\":{\\"\\":{\\"msgid\\":\\"\\",\\"msgstr\\":[\\"Content-Type: text/plain; charset=utf-8\\\\nPlural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"]},\\"test\\":{\\"msgid\\":\\"test\\",\\"msgstr\\":[\\"test translated\\"]}},\\"ctx1\\":{\\"test\\":{\\"msgid\\":\\"test\\",\\"msgctxt\\":\\"ctx1\\",\\"msgstr\\":[\\"test translated\\"]}},\\"ctx2\\":{\\"test\\":{\\"msgid\\":\\"test\\",\\"msgctxt\\":\\"ctx2\\",\\"msgstr\\":[\\"test translated\\"]}}}}"`;

exports[`convert po to js nostrip 1`] = `"{\\"charset\\":\\"utf-8\\",\\"headers\\":{\\"Content-Type\\":\\"text/plain; charset=utf-8\\",\\"Plural-Forms\\":\\"nplurals=2; plural=(n!=1);\\"},\\"translations\\":{\\"\\":{\\"\\":{\\"msgid\\":\\"\\",\\"comments\\":{\\"translator\\":\\"SOME DESCRIPTIVE TITLE\\\\nCopyright (C) YEAR Free Software Foundation, Inc.\\\\nThis file is distributed under the same license as the PACKAGE package.\\\\nFIRST AUTHOR <EMAIL@ADDRESS>, YEAR.\\"},\\"msgstr\\":[\\"Content-Type: text/plain; charset=utf-8\\\\nPlural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"]},\\"test\\":{\\"msgid\\":\\"test\\",\\"comments\\":{\\"reference\\":\\"tests/fixtures/checkTest/check-trans-exist.js:3\\"},\\"msgstr\\":[\\"test translated\\"]}},\\"ctx1\\":{\\"test\\":{\\"msgid\\":\\"test\\",\\"msgctxt\\":\\"ctx1\\",\\"comments\\":{\\"reference\\":\\"tests/fixtures/checkTest/check-trans-exist.js:3\\"},\\"msgstr\\":[\\"test translated\\"]}},\\"ctx2\\":{\\"test\\":{\\"msgid\\":\\"test\\",\\"msgctxt\\":\\"ctx2\\",\\"comments\\":{\\"reference\\":\\"tests/fixtures/checkTest/check-trans-exist.js:3\\"},\\"msgstr\\":[\\"test translated\\"]}}}}"`;

exports[`convert po to js pretty 1`] = `
"{
  \\"charset\\": \\"utf-8\\",
  \\"headers\\": {
    \\"Content-Type\\": \\"text/plain; charset=utf-8\\",
    \\"Plural-Forms\\": \\"nplurals=2; plural=(n!=1);\\"
  },
  \\"translations\\": {
    \\"\\": {
      \\"\\": {
        \\"msgid\\": \\"\\",
        \\"msgstr\\": [
          \\"Content-Type: text/plain; charset=utf-8\\\\nPlural-Forms: nplurals=2; plural=(n!=1);\\\\n\\"
        ]
      },
      \\"test\\": {
        \\"msgid\\": \\"test\\",
        \\"msgstr\\": [
          \\"test translated\\"
        ]
      }
    },
    \\"ctx1\\": {
      \\"test\\": {
        \\"msgid\\": \\"test\\",
        \\"msgctxt\\": \\"ctx1\\",
        \\"msgstr\\": [
          \\"test translated\\"
        ]
      }
    },
    \\"ctx2\\": {
      \\"test\\": {
        \\"msgid\\": \\"test\\",
        \\"msgctxt\\": \\"ctx2\\",
        \\"msgstr\\": [
          \\"test translated\\"
        ]
      }
    }
  }
}"
`;

exports[`should apply compact format 1`] = `"{\\"headers\\":{\\"plural-forms\\":\\"nplurals=2; plural=(n!=1);\\"},\\"contexts\\":{\\"\\":{\\"test\\":[\\"test translated\\"]},\\"ctx1\\":{\\"test\\":[\\"test translated\\"]},\\"ctx2\\":{\\"test\\":[\\"test translated\\"]}}}"`;
